checks.d/disk.py

```diff
@@ -135,13 +135,30 @@ def _exclude_disk_psutil(self, part):
                                           part.fstype == '')) or
                 self._exclude_disk(part.device, part.fstype, part.mountpoint))
 
-    # We don't want all those incorrect devices
     def _exclude_disk(self, name, filesystem, mountpoint):
-        return (((not name or name == 'none') and not self._all_partitions) or
-                name in self._excluded_disks or
-                self._excluded_disk_re.match(name) or
-                self._excluded_mountpoint_re.match(mountpoint) or
-                filesystem in self._excluded_filesystems)
+        """
+        Return True for disks we don't want or that match regex in the config file
+        """
+        name_empty = not name or name == 'none'
+
+        # allow empty names if `all_partitions` is `yes` so we can evaluate mountpoints
+        if name_empty and not self._all_partitions:
+            return True
+        # device is listed in `excluded_disks`
+        elif not name_empty and name in self._excluded_disks:
+            return True
+        # device name matches `excluded_disk_re`
+        elif not name_empty and self._excluded_disk_re.match(name):
```

Comment:
Now that I think about it what I've described is a very minor regression, the behavior you implemented makes more sense.

The worst possible outcome of this "regression" would be that users who have enabled the collection on all partitions end up with slightly more metrics collected. So safe to ignore my comment :)